lib/jekyll/excerpt.rb

```diff
@@ -131,36 +131,44 @@ def render_with_liquid?
     #
     # Returns excerpt String
 
-    LIQUID_TAG_REGEX = %r!{%-?\s*(\w+).+\s*-?%}!m
+    LIQUID_TAG_REGEX = %r!{%-?\s*(\w+).+?\s*-?%}!m
```

Comment:
The capture for this regular expression isn't 100% correct. A liquid tag of `{%highlight%}` will lose the last letter, meaning that the capture group will contain `highligh` instead of `highlight`

I wrote a quick test case for this on Rubular as well, as an extra example: http://rubular.com/r/umPU5g5PiH

Reply:
Thank you for catching this @mattr- :slightly_smiling_face: